Storage-1: Create functionality for 'stock'

AmanEfer opened the pull request:
- created entity, dto, repo, services, mappers and controllers
- tried to connect flyway, but so far to no avail
- created entity, dto, repo, services, mappers and controllers for Product and ProductQuantity
- added saveAllProducts method into ProductService and saveAllStocks method into StockService
- made small changes in other classes

AmanEfer committed:
Storage-1: Create functionality for 'stock'
- created entity, dto, repo, services, mappers and controllers
- tried to connect flyway, but so far to no avail

AmanEfer committed:
Storage-1: Create functionality for 'stock'(temporary commit, this code doesn't work yet)
- created entity, dto, repo, services, mappers and controllers for Product and ProductQuantity
- added saveAllProducts method into ProductService and saveAllStocks method into StockService
- made small changes in other classes

AmanEfer committed:
Storage-1: Create functionality for 'stock'(temporary commit, this code doesn't work yet)
- added migrations by flyway

AmanEfer committed:
Storage-1: Create functionality for 'stock'(temporary commit, this code doesn't work yet)
- update URI in MainController and RestToCrud classes

vicria commented on crud/src/main/java/com/amanefer/crud/models/Product.java:
```diff
@@ -0,0 +1,53 @@
+    private String title;
+
+    @Column(name = "purchase_last_price")
+    private BigDecimal purchaseLastPrice;
```
Попробуй без аннотации колонки запустить

vicria commented on crud/src/main/java/com/amanefer/crud/models/ProductQuantity.java:
```diff
@@ -0,0 +1,33 @@
+@AllArgsConstructor
+@Entity
+@Table(name = "product_quantity", schema = "my_db_bot")
+public class ProductQuantity {
```
А продукта тут не будет ?

vicria commented on crud/src/main/java/com/amanefer/crud/repositories/ProductRepository.java:
```diff
@@ -0,0 +1,16 @@
+public interface ProductRepository extends JpaRepository<Product, String> {
+
+    @Override
+    @NonNull
```
Зачем эта аннотация ?

vicria commented on crud/src/main/java/com/amanefer/crud/services/stock/StockServiceImpl.java:
```diff
@@ -0,0 +1,106 @@
+    @Override
+    public List<StockDto> getAllStocks() {
+
+        return stockRepository.findAll().stream()
```
Может сразу сделать правильный запрос в БД? Вместо фильтрации

AmanEfer committed:
Storage-1: Create functionality for 'stock'(temporary commit, this code doesn't work yet)
- made minor changes: changed annotations, moved default schema to properties, defined findAllByDeletedAtIsNull() method in ProductRepository

AmanEfer committed:
Storage-1: Create functionality for 'stock'(temporary commit, this code doesn't work yet)
- created model classes for all entities;
- updated all controllers, services with these new model classes in mind;
- added some new method in ProductRepository, StockRepository, UserService;
- update saveProduct() in ProductService;
- made minor changes in other classes.

AmanEfer committed:
Storage-1: Create functionality for 'stock'
- added saveAllProducts() method in ProductServiceImpl
- added hardDeleteProduct() in ProductServiceImpl,  add hardDeleteProduct() and softDeleteProduct() in ProductController
- added some new methods in ProductMapper
- made some changes in other classes

AmanEfer committed:
Storage-1: Create functionality for 'stock'
- created saleProducts() method in ProductService and added method of controller for it in ProductController

AmanEfer committed:
Storage-1: Create functionality for 'stock'
- created moveProducts() method in ProductService and added method of controller for it in ProductController

AmanEfer committed:
Storage-1: Create functionality for 'stock'
- created integration tests for saveAllProducts(), saleProducts() and moveProducts() in ProductServiceImpl
- added test-containers dependency in pom(crud), created init.sql file for initialization stock table and created application.yml for run test-containers
- turned off flyway cause it doesn't work(going to fix it later)
- made some minor changes in other classes

vicria commented on crud/pom.xml:
```diff
@@ -41,16 +42,25 @@
             <artifactId>lombok</artifactId>
         </dependency>
 
+<!--        <dependency>-->
```
Почему сразу не удаляешь, а комментишь?

vicria commented on crud/src/main/java/com/amanefer/crud/controllers/ProductController.java:
```diff
@@ -0,0 +1,110 @@
+    }
+
+    @GetMapping("/all")
+    public List<ProductDto> getAllProducts() {
```
Давай к следующему разу попробуем вернуть Page. А на вход будем принимать pagable и фильтр

vicria commented on crud/src/main/java/com/amanefer/crud/controllers/ProductController.java:
```diff
@@ -0,0 +1,110 @@
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public ProductDto saveProduct(@RequestBody ProductDto productDto) {
```
А зачем этот метод?

vicria commented on crud/src/main/java/com/amanefer/crud/entities/Product.java:
```diff
@@ -0,0 +1,46 @@
+
+    private LocalDateTime updatedAt;
+
+    private LocalDateTime deletedAt;
```
Для полноты картины можно добавить еще Transient поле

vicria commented on crud/src/main/java/com/amanefer/crud/entities/Role.java:
```diff
@@ -1,32 +1,26 @@
-@ToString
 @Entity
-@Table(name = "roles")
+@Table(name = "my_role", schema = "my_db_bot")
```
Схему удобнее вынести в проперти файл

vicria commented on crud/src/main/java/com/amanefer/crud/services/user/UserServiceImpl.java:
```diff
@@ -0,0 +1,88 @@
+
+        return userRepository.findAll().stream()
+                .map(userMapper::fromEntityToModel)
+                .map(userMapper::fromModelToDto)
```
А почему в два шага?
